src/App.tsx

```diff
@@ -1,14 +1,58 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect, useMemo } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { Todo } from './types/Todo';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { SortTodos } from './types/SortTodos';
 
 export const App: React.FC = () => {
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [selectTodo, setSelectTodo] = useState<Todo | null>(null);
+  const [selectFilter, setSelectFilter] = useState(SortTodos.All);
+  const [filterField, setFilterField] = useState('');
+  const [isLoading, setIsLoading] = useState(true);
```

Comment:
Better to set `false` as a default value and change it to `true` only before API call and to `false` after API call